core/src/main/scala/org/apache/spark/util/collection/ExternalSorter.scala

```diff
@@ -197,16 +204,24 @@ private[spark] class ExternalSorter[K, V, C](
       while (records.hasNext) {
         addElementsRead()
         kv = records.next()
-        map.changeValue((actualPartitioner.getPartition(kv._1), kv._1), update)
+        val partitionId = actualPartitioner.getPartition(kv._1)
+        map.changeValue((partitionId, kv._1), update)
         maybeSpillCollection(usingMap = true)
+        if (!rowBasedChecksums.isEmpty) {
```

Comment:
super nit:
(here and other places) 

```suggestion
        if (rowBasedChecksums.nonEmpty) {
```